sklearn/metrics/cluster/unsupervised.py

```diff
@@ -305,6 +305,9 @@ def davies_bouldin_score(X, labels):
     labels : array-like, shape (``n_samples``,)
         Predicted labels for each sample.
 
+    pairwise_distances_func : callable
```

Comment:
When I suggested checking if the same issue applied to `davies_bouldin_score`, I'd not realised that it did not work for custom (or precomputed) metric. Should it?

Reply:
Skim reading the original paper indicates that the choice of distance metric is up to the user.  The paper describes required conditions for the metric, which at first glance seem to be standard distance metric requirements and discusses possible choices of metric. For empirical evaluation the paper uses the Euclidean metric.

So it seems like the user should be free to choose their distance metric.

Reply:
If you would like to submit a PR supporting alternative metrics in davied-bouldin, it would be welcome. I'll make an issue.

Reply:
Actually, I take that back. It seems a bit strange that you can use a standard metric, but calculate centroid by the mean of points. I'd need to look at the paper, but ordinarily taking the mean in feature space would mean you can't just use an arbitrary metric. For example, if the distance matrix *was* precomputed, how would you identify the centroid of selected samples?

Reply:
Yes this is an issue. Maybe we can generalise the centroid calculation by using the Frechet/Karcher mean (which I am only just learning about) https://en.wikipedia.org/wiki/Fréchet_mean.

This should work for arbitrary (including precomputed) metrics.

Reply:
Interesting re Fréchet mean (would that require numerical optimisation in the general case?)... But does the Davies-Bouldin paper not comment on the use of mean for non-euclidean metrics?

Reply:
I think it would require numerical optimisation in the general case. I can't find a reference in the paper to non-Euclidean metrics.